lang/cs/Org.Apache.REEF.Common/Properties/AssemblyInfo.DotNet.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Runtime.CompilerServices;
+
+
+// Common and Evaluator share APIs that should not be exposed to the user
+[assembly: InternalsVisibleTo("Org.Apache.REEF.Evaluator")]
```

Comment:
I created a new one for a couple of reasons:
1) The old one has properties that are already defined by the dot net cli toolset. Things like AssemblyTitle, AssemblyDescription -- these are defaulted in the dotnet toolset, so when compiling, these generate build errors because the are defined twice.
2) The InternalsVisible targets the built binary by specifying a name and a key -- since these binaries will be built with different projects, that key will be different. 

Therefore I opted to create a new file altogether. However, an alternative is to have one AssemblyInfo.cs file for both and use the DOT_NET_BUILD preprocessor to enable/disable code that is needed by the dotnet cli tools.

I kind of like that idea -- let me know what you prefer.